rtgui/batchqueuepanel.cc

```diff
@@ -322,6 +337,122 @@ void BatchQueuePanel::setGuiFromBatchState(bool queueRunning, int qsize)
+        auto result = BatchQueue::calcAutoFileNameBase(exampleFilePath);
+        if (!result.empty()) {
+            // The 'd' specifier returns an empty string if N exceeds the number of path elements, so
+            // the largest N that does not return an empty string is the number of elements in exampleFilePath.
+            pathElementCount = n;
+            break;
+        }
+    }
+    // Function inserts examples for a particular specifier, with every valid N value for the
+    // number of elements in the path.
+    auto insertPathExamples = [&buffer, &pos, pathElementCount, exampleFilePath](char letter, int offset1, int mult1, int offset2, int mult2)
+    {
+        for (int n=0; n<pathElementCount; n++) {
+            auto path1 = Glib::ustring::format("%",letter,offset1+n*mult1);
+            auto path2 = Glib::ustring::format("%",letter,offset2+n*mult2);
+            options.savePathTemplate = path1;
+            auto result1 = BatchQueue::calcAutoFileNameBase(exampleFilePath);
+            options.savePathTemplate = path2;
+            auto result2 = BatchQueue::calcAutoFileNameBase(exampleFilePath);
+            pos = buffer->insert_markup(pos, Glib::ustring::format("\n  <tt><b>", path1, "</b> = <b>", path2, "</b> = <i>", result1, "</i></tt>"));
+            if (result1 != result2) {
+                // If this error appears, it indicates a coding error in either BatchQueue::calcAutoFileNameBase
+                // or BatchQueuePanel::populateTemplateHelpBuffer.
+                pos = buffer->insert_markup(pos, Glib::ustring::format(" ", M("QUEUE_LOCATION_TEMPLATE_HELP_RESULT_MISMATCH"), " ", result2));
+            }
+        }
+    };
+    // Example outputs in comments below are for a 4-element path.
+    insertPathExamples('d', pathElementCount, -1, -1, -1);  //   <b>%d4</b> = <b>%d-1</b> = <i>home</i>
+    insertPathExamples('p', 1, 1, -pathElementCount, 1);    //   <b>%p1</b> = <b>%p-4</b> = <i>/home/tom/photos/2010-10-31/</i>
+    insertPathExamples('P', 1, 1, -pathElementCount, 1);    //   <b>%P1</b> = <b>%P-4</b> = <i>2010-10-31/</i>
+    {
+        Glib::ustring fspecifier("%f");
+        options.savePathTemplate = fspecifier;
+        auto result = BatchQueue::calcAutoFileNameBase(exampleFilePath);
+        pos = buffer->insert_markup(pos, Glib::ustring::format("\n  <tt><b>", fspecifier, "</b> = <i>", result, "</i></tt>"));
+    }
+
+    insertTopicHeading(M("QUEUE_LOCATION_TEMPLATE_HELP_RANK_TITLE"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_RANK_BODY"));
+
+    insertTopicHeading(M("QUEUE_LOCATION_TEMPLATE_HELP_SEQUENCE_TITLE"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_SEQUENCE_BODY"));
+
+    insertTopicHeading(M("QUEUE_LOCATION_TEMPLATE_HELP_TIMESTAMP_TITLE"));
+    pos = buffer->insert_markup(pos, M("QUEUE_LOCATION_TEMPLATE_HELP_TIMESTAMP_BODY"));
+    Glib::ustring dateTimeFormatExamples[] = {
```

Comment:
As so many things, this should be `const`.

Reply:
I came up with some searches to find a bunch of them to fix. Will do.